app/src/main/java/org/jellyfin/mobile/utils/LocaleUtils.kt

```diff
@@ -0,0 +1,58 @@
+package org.jellyfin.mobile.utils
+
+import android.app.Activity
+import androidx.lifecycle.lifecycleScope
+import kotlinx.coroutines.launch
+import org.jellyfin.mobile.MainActivity
+import timber.log.Timber
+import java.util.*
+
+private var language: String? = null
+private var country: String? = null
+
+@Suppress("DEPRECATION")
+fun MainActivity.initLocale() {
+    webView.evaluateJavascript("window.localStorage.getItem('${apiClient.currentUserId}-language')") { result ->
+        if (!setLocale(result.unescapeJson(), "userSettings")) {
+            lifecycleScope.launch {
+                if (!setLocale(apiClient.getServerConfiguration()?.uiCulture,"serverSettings")) {
+                    val locale = resources.configuration.locale
+                    setLocale("${locale.language}-${locale.country}", "androidSettings")
+                }
+            }
+        }
+    }
+}
+
+@Suppress("DEPRECATION")
+fun Activity.loadLocale() {
+    val configuration = resources.configuration
+    val locale = Locale(
+        language ?: configuration.locale.language,
+        country ?: configuration.locale.country
+    )
+    Timber.d("loadLocale [class: ${localClassName}, language: '${locale.language}', country: '${locale.country}']")
+    Locale.setDefault(locale)
+    configuration.setLocale(locale)
+    resources.updateConfiguration(configuration, resources.displayMetrics)
+}
```

Comment:
Could you merge this function into the `setLocale` function? Then you could also keep the language/locale variables inside `setLocale`, and wouldn't require the global ones.